src/userobjects/OpenMCTallyEditor.C

```diff
@@ -0,0 +1,161 @@
+/********************************************************************/
+/*                  SOFTWARE COPYRIGHT NOTIFICATION                 */
+/*                             Cardinal                             */
+/*                                                                  */
+/*                  (c) 2024 UChicago Argonne, LLC                  */
+/*                        ALL RIGHTS RESERVED                       */
+/*                                                                  */
+/*                 Prepared by UChicago Argonne, LLC                */
+/*               Under Contract No. DE-AC02-06CH11357               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*             Prepared by Battelle Energy Alliance, LLC            */
+/*               Under Contract No. DE-AC07-05ID14517               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*                 See LICENSE for full restrictions                */
+/********************************************************************/
+
+#ifdef ENABLE_OPENMC_COUPLING
+
+#include "OpenMCTallyEditor.h"
+#include "OpenMCProblemBase.h"
+#include "openmc/tallies/tally.h"
+
+registerMooseObject("CardinalApp", OpenMCTallyEditor);
+
+InputParameters
+OpenMCTallyEditor::validParams()
+{
+  InputParameters params = GeneralUserObject::validParams();
+  params.addParam<bool>("create_tally", false, "Whether to create the tally if it doesn't exist");
+  params.addRequiredParam<int32_t>("tally_id", "The ID of the tally to modify");
+  params.addRequiredParam<std::vector<std::string>>("scores", "The scores to modify in the tally");
+  params.addRequiredParam<std::vector<std::string>>("nuclides",
+                                                    "The nuclides to modify in the tally");
+  params.addRequiredParam<std::vector<std::string>>("filter_ids",
+                                                    "The filter IDs to modify in the tally");
+  params.addParam<bool>(
+      "multiply_density", true, "Whether to multiply the tally by the atom density");
+  params.declareControllable("scores");
+  params.declareControllable("nuclides");
+  params.declareControllable("filter_ids");
+  params.declareControllable("multiply_density");
+  params.addClassDescription("A UserObject for creating and managing OpenMC tallies");
+  return params;
+}
+
+OpenMCTallyEditor::OpenMCTallyEditor(const InputParameters & parameters)
+  : OpenMCUserObject(parameters), _tally_id(getParam<int32_t>("tally_id"))
+{
+  this->initialize();
```

Comment:
Does this `initialize()` have to be here? I'd have thought that MOOSE properly calls the `initialize()` method in the right place. 

Also I guess I'm confused on the comment on line 57 - that function is called from the constructor in practice based on line 51, but the comment makes it sound like it isn't?

Reply:
> Does this `initialize()` have to be here? I'd have thought that MOOSE properly calls the `initialize()` method in the right place.

Hrmm yeah you're probably right. Let me try the tests without that.

Reply:
> Also I guess I'm confused on the comment on line 57 - that function is called from the constructor in practice based on line 51, but the comment makes it sound like it isn't?

Good point. That comment isn't really applicable anymore.

Reply:
What happens if the content inside `initialize()` is just moved into the constructor, and then the `initialize()` method is deleted? Does that also run into issues?

Reply:
If it works, then yes -- `initialize()` is something which MOOSE also calls somewhere in its execution of the `GeneralUserObject` base class, so at a minimum this block of code is being called twice (and possibly cause unexpected things to happen later, depending on what we put in it). Thanks!

Reply:
True true. It doesn't seem as though it was being called at all, but maybe. I didn't dig. The content has been moved into the UO constructors and that seems to work just fine.